lightning/src/ln/channelmanager.rs

```diff
@@ -4983,27 +5078,38 @@ where
 
 		let mut peer_state_lock = peer_state_mutex.lock().unwrap();
 		let peer_state = &mut *peer_state_lock;
-		let ((funding_msg, monitor), chan) =
-			match peer_state.channel_by_id.entry(msg.temporary_channel_id) {
-				hash_map::Entry::Occupied(mut chan) => {
-					(try_chan_entry!(self, chan.get_mut().funding_created(msg, best_block, &self.signer_provider, &self.logger), chan), chan.remove())
+		let (chan, funding_msg, monitor) =
+			match peer_state.inbound_v1_channel_by_id.remove(&msg.temporary_channel_id) {
+				Some(inbound_chan) => {
+					match inbound_chan.funding_created(msg, best_block, &self.signer_provider, &self.logger) {
+						Ok(res) => res,
+						Err((mut inbound_chan, err)) => {
+							// If we get an error at this point we can just forget everything about it
```

Comment:
"it" is ambiguous here. I think you mean the channel but reading the sentence seems to imply that it's referring to the error. I think you still want to convey that this is fine because of the `remove` call above, IIUC.

Reply:
Yeah exactly. I actually think the comment is outdated as I originally didn’t just use the “remove” above. I’ll make it super clear here (or even just before the remove) as I feel it warrants a comment as something “interesting” is happening.